ts/features/walletV3/transaction/components/WalletTransactionHeadingSection.tsx

```diff
@@ -0,0 +1,98 @@
+import { useNavigation } from "@react-navigation/native";
+import Placeholder from "rn-placeholder";
+import React from "react";
+import { View } from "react-native";
+import { Body, H1, IOStyles, VSpacer } from "@pagopa/io-app-design-system";
+import { Psp, Transaction } from "../../../../types/pagopa";
+import { Dettaglio } from "../../../../../definitions/pagopa/Dettaglio";
+import { formatNumberCentsToAmount } from "../../../../utils/stringBuilder";
+import I18n from "../../../../i18n";
+import {
+  WalletTransactionRoutes,
+  WalletTransactionStackNavigation
+} from "../navigation/navigator";
+
+import { WalletTransactionTotalAmount } from "./WalletTransactionTotalAmount";
+import { WalletTransactionDetailsList } from "./WalletTransactionDetailsList";
+
+type Props = {
+  transaction?: Transaction | null;
+  psp?: Psp;
+  loading: boolean;
+};
+
+export const WalletTransactionHeadingSection = ({
+  transaction,
+  psp,
+  loading
+}: Props) => {
+  const navigation = useNavigation<WalletTransactionStackNavigation>();
+
+  const handlePressTransactionDetails = (operationDetails: Dettaglio) => {
+    if (transaction) {
+      navigation.navigate(
+        WalletTransactionRoutes.WALLET_TRANSACTION_OPERATION_DETAILS,
+        {
+          operationDetails,
+          operationName: transaction?.description
+        }
+      );
+    }
+  };
+
+  const FeeAmountSection = () => {
+    if (psp && transaction && transaction.fee && !loading) {
```

Comment:
```suggestion
const hasValidFee = psp!==undefined && transaction?.fee
    if ( hasValidFee && !loading ) {
```

Reply:
Adding this suggestion it will cause an error on the line 46 because for the compiler `transaction.fee` could be null even if we already checked before, it's a type-check needed in this case.

![image](https://github.com/pagopa/io-app/assets/34343582/ab4295c1-be09-433b-afa7-cc5c491090aa)

Reply:
```suggestion
   if (psp &&  transaction?.fee && !loading) {
```
welp yes, typescript does get stupid at times 😅 
sorry about that, this should be all good

Reply:
That's good! 1096aa6b48c45e36cfb4d0e68461146b2f2bf5cd